lib/appsignal/span.ex

```diff
@@ -222,7 +222,7 @@ defmodule Appsignal.Span do
     span
   end
 
-  defp do_set_sample_data(_span, _key, _value), do: nil
```

Comment:
Since all branches of `set_sample_data/3` and its related private functions (`set_sample_data/4`, `do_set_sample_data/3`) end up returning `span` in the end in one way or another, instead of returning `span` from many points in the private functions, we could return `span` only once, at the end of `set_sample_data/3`.